### PR TITLE
Update scripts to build on both Gaea-C5 and C6

### LINKED ISSUE #994: ufs_utils fails to build on Gaea-C5 after OS upgrade
The EPIC SI team has reinstalled spack-stack-1.6.0 at the same location it existed before. 2 spack-stack module versions will need to be increased to build ufs-utils:
stack-intel 2023.1.0 to 2023.2.0
stack-cray-mpich 8.1.25 to 8.1.28

### PR DESCRIPTION
**Pull request by DavidBurrows-NCO**
## DESCRIPTION OF CHANGES: 
The Gaea-C5 OS upgrade changed the version numbers for stack-intel and stack-cray-mpich.  This PR updates the module files to point to the new version numbers.  

In addition, following the ufs-wx-model [PR2448](https://github.com/ufs-community/ufs-weather-model/pull/2448) for Gaea-C6 upgrade, target is set to either gaeac5 and gaeac6, and separate module files are created for each.

## TESTS CONDUCTED: 
Cloned and built on Gaea-C5 and C6.

Standard tests

- [x] Compile branch on all Tier 1 machines using Intel. Done on WCOSS2, Jet, Hera, Orion and Hercules using 82f998c.
- [x] Test the `link_fixdirs.sh` script on all Tier 1 machines. Done on WCOSS2, Jet, Hera, Orion and Hercules using 82f998c.

## DEPENDENCIES:
NA

## DOCUMENTATION:
NA

## ISSUE: 
References NOAA-EMC/global-workflow issue [#3011](https://github.com/NOAA-EMC/global-workflow/issues/3011)

Resolves #994

### CODE REVIEW

**Comment by DavidBurrows-NCO** [comment]
Hello @GeorgeGayno-NOAA I have included updates for building ufs-utils in a container provided by @mark-a-potts.  This is work being coordinated with @aerorahul (@WalterKolczynski-NOAA added since Rahul is on leave) and references NOAA-EMC/global-workflow Issue [#3025](https://github.com/NOAA-EMC/global-workflow/issues/3025).
For links to the fix directories, we don't want to have the fix directories in the container so did not update `link_fixdirs.sh`.
Please let me know if including the container build is okay, and if so, how we should handle `link_fixdirs.sh`.  Maybe a warning message?
Thanks!

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Review by GeorgeGayno-NOAA — CHANGES_REQUESTED** [review]
@DavidBurrows-NCO - I had already approved this PR, but I see you are making more changes. I will re-review when you are done.

**Comment by DavidBurrows-NCO** [comment]
@GeorgeGayno-NOAA Sorry about that! Code is finalized and is consistent with other global-workflow submodules. Thanks.

**Comment by GeorgeGayno-NOAA** [comment]
@DavidBurrows-NCO - 'develop' was updated this morning. Please merge these changes to your branch.

**Comment by DavidBurrows-NCO** [comment]
@GeorgeGayno-NOAA I updated my fork and tested a clone/build on C6 successfully.

**Comment by GeorgeGayno-NOAA** [comment]
@DavidBurrows-NCO - 'develop' was updated again. Please merge these updates to your branch and retest. Then, I will merge your changes. Thanks.

**Comment by DavidBurrows-NCO** [comment]
@GeorgeGayno-NOAA All set. Thanks.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

File: modulefiles/build.container.intel.lua [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+load(pathJoin("sp", sp_ver))
+
+w3emc_ver=os.getenv("w3emc_ver") or "2.10.0"
+load(pathJoin("w3emc", w3emc_ver))
+
+sfcio_ver=os.getenv("sfcio_ver") or "1.4.1"
+load(pathJoin("sfcio", sfcio_ver))
+
+sigio_ver=os.getenv("sigio_ver") or "2.3.2"
+load(pathJoin("sigio", sigio_ver))
```
Comment (aerorahul) [inline modulefiles/build.container.intel.lua]: Why is this needed in a container?

File: modulefiles/build.container.intel.lua [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+load(pathJoin("libpng", png_ver))
+
+netcdf_c_ver=os.getenv("netcdf_c_ver") or "4.9.2"
+load(pathJoin("netcdf-c", netcdf_c_ver))
+
+netcdf_fortran_ver=os.getenv("netcdf_fortran_ver") or "4.6.1"
+load(pathJoin("netcdf-fortran", netcdf_fortran_ver))
+
+nccmp_ver=os.getenv("nccmp_ver") or "1.9.0.1"
+load(pathJoin("nccmp", nccmp_ver))
```
Comment (aerorahul) [inline modulefiles/build.container.intel.lua:47]: Which utility is using this at build time?

File: modulefiles/build.container.intel.lua [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+esmf_ver=os.getenv("esmf_ver") or "8.6.1"
+load(pathJoin("esmf", esmf_ver))
+
+nco_ver=os.getenv("nco_ver") or "5.1.6"
+load(pathJoin("nco", nco_ver))
```
Comment (aerorahul) [inline modulefiles/build.container.intel.lua:53]: Same q. as nccmp

File: modulefiles/build.container.intel.lua [hunk 4]
```diff
@@ -0,0 +1,61 @@
+help([[
+Load environment to compile UFS_UTILS in a container using Intel
+]])
+
+hpss_ver=os.getenv("hpss_ver") or ""
+--load(pathJoin("hpss", hpss_ver))
```
Comment (aerorahul) [inline modulefiles/build.container.intel.lua]: Why is hpss_ver here if its not loaded or used at buid time?

**Review by aerorahul — COMMENTED** [review]
while reviewing gfs-utils, I came across this PR as well for Gaea C5 C6.  Some comments.

File: modulefiles/build.container.intel.lua (hunk 2 above)
Comment (GeorgeGayno-NOAA) [reply modulefiles/build.container.intel.lua:47]: It is not needed for building UFS_UTILS. It is used by the regression testing. UFS_UTILS does not have separate build and run time modules.

File: modulefiles/build.container.intel.lua (hunk 3 above)
Comment (GeorgeGayno-NOAA) [reply modulefiles/build.container.intel.lua:53]: Same answer as nccmp.

File: modulefiles/build.container.intel.lua (hunk 4 above)
Comment (mark-a-potts) [reply modulefiles/build.container.intel.lua]: I removed hpss and sfcio from the container modules.

**Comment by GeorgeGayno-NOAA** [comment]
@aerorahul, thanks for your review. Any further comments?

**Review by aerorahul — APPROVED** [review]
lgtm.

**Comment by aerorahul** [comment]
> @aerorahul, thanks for your review. Any further comments?

nopes. look good. thanks for checking.

**Comment by DavidBurrows-NCO** [comment]
Thanks @GeorgeGayno-NOAA @aerorahul @mark-a-potts Have a good weekend.
